src/App.vue

```diff
@@ -519,16 +525,26 @@ export default class App extends Mixins(CommonMixin, DateMixin, FilingTemplateMi
     this.updateCurrentJsDateId = setInterval(this.updateCurrentJsDate, 60000)
 
     // add handler to prompt user if there are changes, before unloading this page
-    window.onbeforeunload = (event) => {
+    // NB: this is not reliably fired by browsers, especially on mobile
+    window.onbeforeunload = (event: BeforeUnloadEvent) => {
       if (this.getHaveChanges) {
         event.preventDefault()
         // NB: custom text is not supported in all browsers
         event.returnValue = 'You have unsaved changes. Do you want to exit?'
       }
     }
 
+    // add handler to delete unsaved documents when page is hidden / unloaded
+    // (don't use window.onload as it's deprecated)
+    // NB: this is not reliably fired by browsers, especially on mobile
+    window.onpagehide = async (event: PageTransitionEvent) => {
+      if (!event.persisted) {
+        await this.deleteUnsavedDocuments()
```

Comment:
This await does nothing. This event handler is not actually async. 😞 

There is also no way to run something specific if the user clicks to leave this site in the BeforeUnload event handler above (which is also synchronous).